File: Android/WorldTrivia/app/src/main/java/com/example/worldtrivia/MainActivity.java
package com.example.worldtrivia;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;

import com.example.worldtrivia.model.QuizManager;

public class MainActivity extends AppCompatActivity {
    private QuizManager quizManager;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        quizManager = new QuizManager(this);
        quizManager.getQuestions(this);
    }
}

File: android/VidyaKhoj/app/src/main/java/com/sidharth/vidyakhoj/ui/MainActivity.kt
package com.sidharth.vidyakhoj.ui

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.lifecycleScope
import com.sidharth.vidyakhoj.databinding.ActivityMainBinding
import com.sidharth.vidyakhoj.service.RefreshDataService
import dagger.hilt.android.AndroidEntryPoint
import kotlinx.coroutines.delay
import kotlinx.coroutines.launch

@AndroidEntryPoint
class MainActivity : AppCompatActivity() {
    private val mainBinding: ActivityMainBinding by lazy {
        ActivityMainBinding.inflate(layoutInflater)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(mainBinding.root)
        lifecycleScope.launch {
            Intent(this@MainActivity, RefreshDataService::class.java).apply {
                action = RefreshDataService.Action.START.toString()
                delay(10000)
                startService(this)
            }
        }
    }
}